request-handler/src/main/java/au/org/aodn/aws/wps/operation/ExecuteOperation.java

```diff
@@ -68,23 +74,23 @@ public Object execute(Properties config) {
         String jobId = result.getJobId();
 
         System.out.println("Job submitted.  Job ID : " + jobId);
-        //TODO: create job status document - status = submitted!
+        StatusCreator statusUpdater = new StatusCreator("http://" + statusLocationBase + "/" + jobId + "/" + statusFileName, jobId);
+        ExecuteResponse response = statusUpdater.createResponseDocument(EnumStatus.ACCEPTED);
 
-        String statusLocation = statusLocationBase + jobId + "/" + statusFileName;
+        String document = StatusCreator.createXmlDocument(response);
 
-        ExecuteResponse response = new ExecuteResponse();
-        response.setStatusLocation(statusLocation);
-        StatusType status = new StatusType();
+        AmazonS3 cl = AmazonS3ClientBuilder.defaultClient();
 
         try {
-            status.setCreationTime(DatatypeFactory.newInstance().newXMLGregorianCalendar());
-        } catch (DatatypeConfigurationException e) {
-            throw new RuntimeException(e);
+            cl.putObject(
+                    new PutObjectRequest(statusLocationBase, jobId + "/" + statusFileName, new StringInputStream(document), new ObjectMetadata())
+                            .withCannedAcl(CannedAccessControlList.PublicRead));
+        } catch (UnsupportedEncodingException e) {
+            //TODO: if uploading status file fails should we return a failed status?
+            e.printStackTrace();
```

Comment:
We could return a generic OGC exception instead e.g. http://schemas.opengis.net/wps/1.0.0/examples/90_wpsExceptionReport.xml except using a NoApplicableCode exceptionCode and include an appropriate exception message.